src/univariate/continuous/weibull.jl

```diff
@@ -139,3 +139,55 @@ end
 #### Sampling
 
 rand(rng::AbstractRNG, d::Weibull) = xv(d, randexp(rng))
+
+#### Fit model
+
+"""
+    fit_mle(::Type{<:Weibull}, x::AbstractArray{<:Real}; 
+    alpha0::Real = 1, maxiter::Int = 1000, tol::Real = 1e-16)
+
+Compute the maximum likelihood estimate of the [`Weibull`](@ref) distribution with Newton's method.
+"""
+function fit_mle(::Type{<:Weibull}, x::AbstractArray{<:Real};
+    alpha0::Real = 1, maxiter::Int = 1000, tol::Real = 1e-16)
+
+    N = 0
+
+    lnx = map(log, x)
+    lnxsq = lnx.^2
+    mean_lnx = mean(lnx)
+    n = length(x)
+
+    # first iteration outside loop, prevents type instabililty in α, ϵ
+
+    xpow0 = x.^alpha0
+    sum_xpow0 = sum(xpow0)
+    dot_xpowlnx0 = dot(xpow0, lnx)
+
+    fx = dot_xpowlnx0 / sum_xpow0 - mean_lnx - 1 / alpha0
+    ∂fx = (-dot_xpowlnx0^2 + sum_xpow0 * dot(lnxsq, xpow0)) / (sum_xpow0^2) + 1 / alpha0^2
+
+    α = alpha0 - fx / ∂fx
+
+    ϵ = abs(α - alpha0)
```

Comment:
This is just `abs(fx / ∂fx)`, so it might be easier to compute `Δα = fx / ∂fx` and use it both to update `α` and to compute `ϵ` (also below).

Reply:
Good catch, I'll update that